api/src/test/java/com/example/api/service/group/GroupServiceTests.java

```diff
@@ -94,4 +103,46 @@ public void getEmptyGroupCodeListTest() {
         Assertions.assertEquals(result, List.of());
 
     }
+
+    @Test
+    public void getGroupUserListTest() throws EntityNotFoundException {
+        // given
+        User user1 = new User();
+        User user2 = new User();
+
+        group1.setUsers(List.of(user1, user2));
+        when(groupRepo.findById(group1.getId())).thenReturn(Optional.ofNullable(group1));
```

Comment:
No tak jak pisałem Ci na dc dzisiaj wydaje mi się, że te testy do serwisu są błędne, ponieważ nie sprawdzasz tu co sie dzieje w serwisach, tylko to co tak naprawdę robią repozytoria (a do nich bedą osobne testy).

Tutaj:
1. Podajesz, że gdy wykonasz `groupRepo.findById(group1.getId())` to ma Ci zwrócić group1.
2. (Niżej) Wykonujesz `groupService.getGroupUserList(group1.getId())`
i sprawdzasz czy return zgadza się z tym co oczekujesz.
3. To działa tak, że ustawiasz co ma Ci zwrócić, a potem sprawdzasz czy serio Ci to zwraca (chociaż sam to ustawiłeś). Znaczny no tu jest troche inaczej bo jeszcze na tym group1 robisz jakieś operacje, ale sam zamysł pozostaje ten sam.

Wrzuciłem tego mojego PR'a testy, których ja sie uczyłem. Uczyłem się tego jakoś daaawno temu między innymi z tego filmiku: https://www.youtube.com/watch?v=Geq60OVyBPg&t=4091s, ale to bardzo w podobny sposób testujemy w pracy.

Głównie chodzi o to, aby to co nam zwracają repozytoria w serwisie sobie ustawiać właśnie przez when().thenReturn, albo jak ja robilem given().willReturn. Następnie za pomocą ArgumentCaptora łapiemy sobie te argumenty, które były wykorzystane w tej funkcji i sprawdzamy czy się zgadzaja z tymi co ześmy dali jako prawdziwe argumenty. 

Jeżeli chodzi o to co zwraca serwis to, jeżeli jest to np. repo.save(entity). To wtedy nas nie interesuje za bardzo to, ponieważ zwraca coś z repa i to przy testach repo będziemy sprawdzać czy rzeczywiscie epo.save(entity) zwraca to dobre entity. A w testach serwisu sprawdzamy, czy rzeczywiscie te obiekty i argumenty, które tam złapiemy to rzeczywiście są te, które tam daliśmy lub stworzyliśmy.

Oczywiście to nie jest zawsze i w zależnosci od serwisu się powinno robić testy, bo niektóre zwracają co innego i wtedy trzeba posprwadzać odpowiednio wszystko (dopóki wciąż należy to dp zakresu zadań danego serwisu)

Troche się rozpisałem, ale może to rozwieje wątpliwości - jak masz uwagi, bo gadam głupoty, albo chcesz żebym Ci wytlumaczył to hit me up!